### PR TITLE
Feat/puzzle and crossword

### PR DESCRIPTION
**Pull request by wk981**
Set up puzzles and crosswords pages. I have also set up a dev deployment.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -14,6 +14,9 @@ import QuizPage from "./pages/QuizPage";
 import NewsPage from "./pages/NewsPage";
 import { ToastProvider } from "./context/ToastContext";
 import { ConfigProvider } from "antd";
+// import { PuzzlesPage } from "./pages/PuzzlePage";
```
Comment (matthewdinata) [inline src/App.tsx:24]: pls delete this comment

File: src/pages/PuzzlePage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    // get the value of ?game=...
+    const gameParams = searchParams.get('game');
+    return (
+        <div className="w-full flex overflow-hidden min-w-[200px]">
+            {/* Check searchParams */}
+            {/* Is this the best practice? If not. TODO: Follow Best Practice */}
+            {gameParams === "timeTrek" && <TimeTrekFrame />}
+            {gameParams === "timeTrivia" && <TimeTriviaFrame />}
+            {!gameParams &&
```
Comment (matthewdinata) [inline src/pages/PuzzlePage/index.tsx:41]: see comments above

File: src/pages/PuzzlePage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { MultiLinkLayer } from "@/components/Layout";
+import { MockGameDataTypes } from "@/types/linkToGame";
+import { useSearchParams } from "react-router-dom";
+import { TimeTrekFrame } from "./components/TimeTrek";
+import { TimeTriviaFrame } from "./components/TimeTrivia";
+// import { useState } from "react";
+const gameListData: MockGameDataTypes[] = [{
```
Comment (matthewdinata) [inline src/pages/PuzzlePage/index.tsx:7]: more more modular approach, also store the component in the gameListData, and then render the game according to the selected game, 

maybe you can do something like: 
```
...
linkTo: {
     timeTrek: {
          id: 1,
          title: ..
          component: ..
     },
     ...
}
```


render `gameListData.linkTo.gameParams.component`

File: src/pages/CrosswordPage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { MultiLinkLayer } from "@/components/Layout";
+import { MockGameDataTypes } from "@/types/linkToGame";
+import { useSearchParams } from "react-router-dom";
+import { CyberSafeCrosswordsFrame } from "./components/CyberSafeCrossWords";
+import { CyberWordHunterFrame } from "./components/CyberWordHunt";
```
Comment (matthewdinata) [inline src/pages/CrosswordPage/index.tsx:5]: check similar comment on puzzles page

File: src/pages/CrosswordPage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+export const CrosswordsPage = () => {
+    // TODO, use setgameList if we want to implement the completion of game
+    // const [gameList, _setGameList] = useState<MockGameLinkDataTypes[]>(gameListData);
+    // Get the searchparams from the URL
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    // get the value of ?game=...
+    const gameParams = searchParams.get('game');
+    console.log(gameParams);
```
Comment (matthewdinata) [inline src/pages/CrosswordPage/index.tsx:47]: clean all console logs

File: src/components/Layout/index.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import { FC, ReactNode} from "react"
+import { IFrameRendererProps, MultiLinkLayerProps } from "./types"
+import { useLoadedIframe } from "@/hooks/useLoadedIframe"
+import { Spinner } from "../Spinner"
+import { SetURLSearchParams } from "react-router-dom"
+
+export const MultiLinkLayer = ({ title, linkToArray,setSearchParams}: MultiLinkLayerProps) => {
```
Comment (matthewdinata) [inline src/components/Layout/index.tsx:7]: can we have a rounded border and a drop shadow to make the games div render a bit smoother?

Comment (wk981) [reply src/components/Layout/index.tsx:7]: I have implemented it. Most of the game div is able to adapt it however 
```
<TimeTriviaFrame />
```

does not adapt to it prob due to its provider

**Review by matthewdinata — APPROVED** [review]

**Pull request merged**